conans/client/configure_environment.py

```diff
@@ -62,7 +62,9 @@ def _gcc_env(self):
         lib_paths = " ".join(["-L%s" % lib for lib in self._deps_cpp_info.lib_paths])
         ldflags = 'LDFLAGS="%s %s %s %s $LDFLAGS"' % (lib_paths, archflag,
                                                       exe_linker_flags, shared_linker_flags)
-        debug = "-g" if self.build_type == "Debug" else "-s -DNDEBUG"
+        debug = "-g" if self.build_type == "Debug" else "-DNDEBUG"
+        if self.compiler == "gcc" and self.build_type == "Debug":
```

Comment:
Wouldn't it be ``self.build_type != "Debug"``?
CI is broken, I am fixing it, will send you a PR